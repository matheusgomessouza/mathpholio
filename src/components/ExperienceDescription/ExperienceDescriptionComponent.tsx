import * as interfaces from "@/types/interfaces";

export default function ExperienceDescriptionComponent({
  jobTitle,
  company,
  locality,
  period,
  children,
}: interfaces.ExperienceDescriptionComponentProps) {
  return (
    <article className="relative z-20 w-full py-8">
      <section className="flex flex-col items-center justify-between md:flex-row xl:justify-start xl:gap-8 2xl:justify-between">
        <div className="flex w-full items-center justify-between md:hidden">
          <h3>{jobTitle}</h3>
          <h3 className="underline md:hidden">{company}</h3>
        </div>
        <h3 className="hidden lg:flex 2xl:text-xl">{jobTitle}</h3>
        <h3 className="hidden underline lg:flex 2xl:text-xl">{company}</h3>
        <section className="flex w-full items-center justify-between gap-8 md:w-auto md:justify-end 2xl:text-xl">
          <p>{locality}</p>
          <h3>{period}</h3>
        </section>
      </section>
      <ul className="mt-4 list-disc pl-8">{children}</ul>
    </article>
  );
}
