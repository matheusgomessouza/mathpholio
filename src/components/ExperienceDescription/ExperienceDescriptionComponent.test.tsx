/**
 * @vitest-environment jsdom
 */

import React from "react";
import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import { ExperienceDescriptionComponent } from "./ExperienceDescriptionComponent";

describe("ExperienceDescriptionComponent", () => {
  const defaultProps = {
    jobTitle: "Software Engineer",
    company: "Tech Corp",
    locality: "Remote",
    period: "2020 - 2023",
    children: (
      <>
        <li>Developed cool features</li>
        <li>Improved performance</li>
      </>
    ),
  };

  it("renders job title and company in mobile view", () => {
    render(<ExperienceDescriptionComponent {...defaultProps} />);
    // Both jobTitle and company should be rendered at least once
    expect(screen.getAllByText(defaultProps.jobTitle).length).toBeGreaterThan(
      0
    );
    expect(screen.getAllByText(defaultProps.company).length).toBeGreaterThan(0);
  });

  it("renders locality and period", () => {
    render(<ExperienceDescriptionComponent {...defaultProps} />);
    expect(screen.queryAllByText(defaultProps.locality)).not.toBeNull();
    expect(screen.queryAllByText(defaultProps.period)).not.toBeNull();
  });

  it("renders children as list items", () => {
    render(<ExperienceDescriptionComponent {...defaultProps} />);
    expect(screen.getAllByText("Developed cool features")).not.toBeNull();
    expect(screen.getAllByText("Improved performance")).not.toBeNull();
    const list = screen.getAllByRole("list");
    expect(list).not.toBeNull();
    expect(screen.getAllByRole("listitem").length).toBe(6);
  });

  it("matches snapshot", () => {
    const { container } = render(
      <ExperienceDescriptionComponent {...defaultProps} />
    );
    expect(container).toMatchSnapshot();
  });
});
