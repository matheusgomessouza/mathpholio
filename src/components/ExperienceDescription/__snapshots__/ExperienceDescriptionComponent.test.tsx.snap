// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ExperienceDescriptionComponent > matches snapshot 1`] = `
<div>
  <article
    class="relative z-20 w-full py-8"
  >
    <section
      class="flex flex-col items-center justify-between md:flex-row xl:justify-start xl:gap-8 2xl:justify-between"
    >
      <div
        class="flex w-full items-center justify-between md:hidden"
      >
        <h3>
          Software Engineer
        </h3>
        <h3
          class="underline md:hidden"
        >
          Tech Corp
        </h3>
      </div>
      <h3
        class="hidden lg:flex 2xl:text-xl"
      >
        Software Engineer
      </h3>
      <h3
        class="hidden underline lg:flex 2xl:text-xl"
      >
        Tech Corp
      </h3>
      <section
        class="flex w-full items-center justify-between gap-8 md:w-auto md:justify-end 2xl:text-xl"
      >
        <p>
          Remote
        </p>
        <h3>
          2020 - 2023
        </h3>
      </section>
    </section>
    <ul
      class="mt-4 list-disc pl-8"
    >
      <li>
        Developed cool features
      </li>
      <li>
        Improved performance
      </li>
    </ul>
  </article>
</div>
`;
